sympy/stats/tests/test_rv.py

```diff
@@ -107,6 +107,14 @@ def test_IndependentProductPSpace():
 def test_E():
     assert E(5) == 5
 
+def test_H():
+    X = Normal('X', 0, 1)
+    D = Die('D', sides = 4)
+    G = Geometric('G', 0.5)
+    assert str(H(X, X > 0)) == '-log(2)/2 + 1/2 + log(pi)/2'
```

Comment:
It should not be necessary to use strings for comparison if you write `S(1)/2` instead of `1/2`.

Reply:
Thanks for the suggestion. I have removed the string comparison.